test/controllers/users.test.js

```diff
@@ -1,99 +1,226 @@
 const request = require('supertest');
 
 const app = require('../../app.js');
-const userFactory = require('../factory/users');
+const { createUser } = require('../factory/users');
+const { AUTH_ERROR, CONFLICT_ERROR, SCHEMA_ERROR } = require('../../app/errors');
 const {
-  CONFLICT_CODE,
-  CREATED_CODE,
-  DEFAULT_HEADER,
-  UNPROCESSABLE_ENTITY_CODE,
-  USERS_PATH
-} = require('../constants/');
+  statusCode: { CONFLICT_CODE, CREATED_CODE, NOT_AUTH_CODE, OK_CODE, UNPROCESSABLE_ENTITY_CODE }
+} = require('../../app/constants/');
 const {
   conflictErrorResponse,
   correctWoloxEmail,
+  mockUser,
   randomMissingParams,
-  user: userTestInfo,
+  signInEmpty,
+  userAuthResponse,
   userCreatedResponse,
+  wrongCredentials,
+  wrongEmail,
+  wrongPassword,
   wrongPasswords,
   wrongWoloxEmail
 } = require('../fixtures/users');
 
-describe(`SignUp endpoint POST ${USERS_PATH}`, () => {
-  describe('satisfactory cases - suite', () => {
-    test('success json reponse (when users was created)', () =>
-      request(app)
-        .post(USERS_PATH)
-        .set(DEFAULT_HEADER)
-        .send(userTestInfo)
-        .then(res => expect(res.body).toStrictEqual(userCreatedResponse)));
-    test('user successfully created', () =>
-      request(app)
-        .post(USERS_PATH)
-        .set(DEFAULT_HEADER)
-        .send(userTestInfo)
-        .then(res => expect(res.statusCode).toBe(CREATED_CODE)));
-    test.each(correctWoloxEmail)('user created with a valid wolox email: %p', randomTestData =>
-      request(app)
-        .post(USERS_PATH)
-        .set(DEFAULT_HEADER)
-        .send(randomTestData)
-        .then(res => expect(res.statusCode).toBe(CREATED_CODE))
-    );
-  });
-  describe('failed cases - suite', () => {
-    test('failed with an email already exists', () =>
-      userFactory
-        .create()
-        .then(() =>
-          request(app)
-            .post(USERS_PATH)
-            .set(DEFAULT_HEADER)
-            .send(userTestInfo)
-        )
-        .catch(res => expect(res.statusCode).toBe(CONFLICT_CODE)));
-    test('conflict error json reponse (when email already)', () =>
-      userFactory
-        .create()
-        .then(() =>
-          request(app)
-            .post(USERS_PATH)
-            .set(DEFAULT_HEADER)
-            .send(userTestInfo)
-        )
-        .catch(res => expect(res.body).toStrictEqual(conflictErrorResponse)));
-    test.each(randomMissingParams)('failed with an or more missing parameter: %p', randomTestData =>
-      userFactory
-        .create()
-        .then(() =>
+const DEFAULT_HEADER = ['Accept', 'application/json'];
+const SIGN_IN_PATH = '/users/sessions';
+const USERS_PATH = '/users';
+
+describe(`Sign Up endpoint POST ${USERS_PATH} - suite`, () => {
+  let response = {};
+  describe('create a new user', () => {
+    beforeAll(async () => {
```

Comment:
Maybe this describe should be inside in the next describe  `'satisfactory cases - suite'`. What do you think?